python/ml4ir/base/features/feature_config.py

```diff
@@ -465,13 +473,11 @@ def parse_config(
     tfrecord_type: str, feature_config, logger: Optional[Logger] = None
 ) -> FeatureConfig:
     if feature_config.endswith(".yaml"):
-        feature_config = file_io.read_yaml(feature_config)
-        if logger:
-            logger.info("Reading feature config from YAML file : {}".format(feature_config))
+        feature_config = file_io.read_yaml(feature_config, log=logger)
     else:
         feature_config = yaml.safe_load(feature_config)
-        if logger:
-            logger.info("Reading feature config from YAML string")
+    if logger:
+        logger.info("Feature Config \n{}".format(json.dumps(feature_config, indent=4)))
```

Comment:
btw these are very nice, but not related to the functionality this PR adds!
thanks for the effort though! We could also consider this as logger debug, as it is often long

Reply:
True. Can change.